docs/resources/scenarios/analyze-web-downloads.md

```diff
@@ -0,0 +1,195 @@
+        conditionalFormatTrue.cellValue.rule = {
+          formula1: "=TRUE",
+          operator: "EqualTo"
+        };
+        conditionalFormatFalse.cellValue.format.fill.color = "#F8CCAD";
+        conditionalFormatFalse.cellValue.rule = {
+          formula1: "=FALSE",
+          operator: "EqualTo"
+        };
+      }
+
+      // Adds the summary data to the current sheet and to the summary table.
+      function addSummaryData() {
+        // Add a summary row and table.
+        let summaryHeader = [["Year", "Week", "US", "Other"]];
+        let countTrueFormula =
+          "=COUNTIF(" + isUSColumn.address + ', "=TRUE")/' + (newCol.length - 1);
+        let countFalseFormula =
+          "=COUNTIF(" + isUSColumn.address + ', "=FALSE")/' + (newCol.length - 1);
+
+        let summaryContent = [
+          [
+            '=TEXT(A2,"YYYY")',
+            '=TEXTJOIN(" ", FALSE, "Wk", WEEKNUM(A2))',
+            countTrueFormula,
+            countFalseFormula
+          ]
+        ];
+        let summaryHeaderRow = context.workbook.worksheets
+          .getActiveWorksheet()
+          .getRange("H1:K1");
+        let summaryContentRow = context.workbook.worksheets
+          .getActiveWorksheet()
+          .getRange("H2:K2");
+        summaryHeaderRow.values = summaryHeader;
+        summaryContentRow.values = summaryContent;
+        let formats = [[".000", ".000"]];
+        summaryContentRow
+          .getOffsetRange(0, 2)
+          .getResizedRange(0, -2).numberFormat = formats;
+      }
+    }
+
+    // Translate an IP address into an integer.
+    function ipInt(ip) {
+      // Split the IP address into octets.
+      let octets = ip.split(".");
+
+      // Create a number for each octet and do the math to create the integer value of the IP address.
+      let fullNum =
+        // Define an arbitrary number for the last octet.
+        111 +
+        parseInt(octets[2]) * 256 +
+        parseInt(octets[1]) * 65536 +
+        parseInt(octets[0]) * 16777216;
+      return fullNum;
+    }
+
+    // Return the row number where the ip address is found.
+    function findIP(ipLookupTable: number[][], n: number) {
+      for (let i = 0; i < ipLookupTable.length; i++) {
+        if (ipLookupTable[i][0] <= n && ipLookupTable[i][1] >= n) {
+          return i;
+        }
+      }
+      return -1;
+    }
+    ```
+
+5. Rename the script to **Analyze Web Downloads** and save it.
+
+## Running the script
+
+Run the **Analyze Web Downloads** script on one of the **Week\*\*** worksheets. This will apply the conditional formatting and location labelling on the current sheet. It also updates the **Summary** worksheet.
+
+### Before running the script
+
+![A worksheet that shows raw web traffic data.](../../images/scenario-analyze-web-downloads-before.png)
+
+### After running the script
+
+![A worksheet that shows formatted IP location information with the previous web traffic rows.](../../images/scenario-analyze-web-downloads-after.png)
```

Comment:
FWIW, the "After" data looks a bit odd to me...I'd expect all the original data to be preserved and new data to be purely additive (not overwriting any of the original data). It feel weird that:
- the original value of the column H header in row 1 has been overwritten (Referrer -> Year),
- the value of column H in the first row of data (row 2) has been overwritten by the new data (http://www.... -> 2020), while original values in column H starting with the second row of data (row 3) are preserved